pytask_io/pytask_io.py

```diff
@@ -154,21 +160,16 @@ def stop(self) -> None:
 
         :return: None
         """
-        # stop event loop
-        current_loop = asyncio.get_event_loop()
-        current_loop.call_soon_threadsafe(current_loop.stop)
-        self.main_loop.call_soon_threadsafe(self.main_loop.stop)
-        
-        self.loop_thread.join()
-
-    def _run_event_loop(self) -> None:
+        self.queue_client.lpush(_QUEUE_NAME, QueueActions.STOP.name)
```

Comment:
This needs to be agnostic to the queue framework so will put in a utils functions..